openff/cli/generate_conformers.py

```diff
@@ -0,0 +1,325 @@
+from typing import List, Optional, Tuple
+
+# from rdkit.Chem import rdMolAlign
+import numpy as np
+from openforcefield.topology.molecule import Molecule, UndefinedStereochemistryError
+from openforcefield.typing.engines.smirnoff import ForceField
+from openforcefield.utils.toolkits import ToolkitRegistry
+from simtk import openmm, unit
+
+
+def generate_conformers(
+    molecule: str,
+    registry: ToolkitRegistry,
+    forcefield: str,
+    constrained: bool = False,
+    prefix: Optional[str] = None,
+) -> List[Molecule]:
+
+    ff_name = forcefield
+    if constrained:
+        split_name = ff_name.split("-")
+        split_name[0] = "openff_unconstrained"
+        ff_name = "-".join(split_name)
+    if not ff_name.endswith(".offxml"):
+        ff_name += ".offxml"
+    ff = ForceField(ff_name)
+
+    file_format = molecule.split(".")[-1]
+
+    # TODO: This may not preserve order of loading molecules in
+    ambiguous_stereochemistry = False
+    try:
+        raw_mols = registry.call(
+            "from_file",
+            molecule,
+            file_format=file_format,
+        )
+    except UndefinedStereochemistryError:
+        ambiguous_stereochemistry = True
+        raw_mols = registry.call(
+            "from_file",
+            molecule,
+            file_format=file_format,
+            allow_undefined_stereo=True,
+        )
+
+    # When failing to parse molecules (i.e. attempting to read MOL2 with
+    # RDKit, which is not supported) the toolkit can return an empty
+    # list instead of raising a specific exception
+    if raw_mols == []:
+        from openff.cli.utils.exceptions import MoleculeParsingError
+
+        raise MoleculeParsingError(toolkit_registry=registry, filename=molecule)
+
+    mols = []
+    for i, mol in enumerate(raw_mols):
+        if prefix is not None:
+            mol.name = prefix
+        elif not mol.name:
+            mol.name = "molecule"
+        if len(raw_mols) > 1:
+            mol.name += str(i)
+        mols.append(mol)
+
+    # TODO: How to handle names of different stereoisomers? Just act like they're different conformers?
+    if ambiguous_stereochemistry:
+        mols_with_unpacked_stereoisomers = []
+        for mol in mols:
+            # TODO: This is a brute-force approach, it would be better to check stereo
+            #  without needing to call enumerate_stereoisomers
+            stereoisomers = mol.enumerate_stereoisomers()
+            if stereoisomers:
+                for i, iso in enumerate(stereoisomers):
+                    iso.name = mol.name + "_stereoisomer" + str(i)
+                    mols_with_unpacked_stereoisomers.append(iso)
+            else:
+                mols_with_unpacked_stereoisomers.append(mol)
+        mols = mols_with_unpacked_stereoisomers
+
+    # If no conformers were found (i.e. SMI files), generate some
+    # TODO: How many should be generated?
+    # TODO: If 1 or a few conformers are found, should more be generated?
```

Comment:
I think the proper behavior here would be to _always_ generate conformers. If we can keep the input conformer(s) as a starting point for minimizations, that's a plus, but not absolutely necessary.

Folks who want to do something with some pre-existing conformers should use our `conformer_energies` tool.